### DISCUSSION TITLE
Unable to run tests via Xcode (via `make test` they run perfectly)

### DISCUSSION THREAD

**Discussion by dcacenabes** [swiftyfinch/Rugby#394] [Q&A]
Hi!
I have a doubt. I am trying to run the test suite from Xcode on a freshly cloned Rugby repository and I get compilation errors:
`@testable import Rugby` line throws error: `No such module 'Rugby'`

Steps to reproduce:
1. Clone the project from Github
2. Open Package.swift in Xcode (I tried both 15.3 and 15.4 with same result)
3. Use command+u key combination (or select run tests on the Xcode menu)

Expected output:
Tests build and run

Actual output:
Compilation error.

I am sure there must be some basic setup that I am missing. Would love to be able to run the tests from Xcode to easily debug new tests that I am working on. Thanks in advance!

**Comment by dcacenabes** [comment]
Small update, by editing the schemes and adding the scheme for RugbyTests and FoundationTests, I am now able of running FoundationTests correctly from Xcode, but RugbyTests still fails with the same compiler error. 
<img width="726" alt="Screenshot 2024-05-17 at 10 34 38" src="https://github.com/swiftyfinch/Rugby/assets/3796899/8e4730fc-af81-434e-ad70-7ea33c4a02a4">

**Reply by swiftyfinch** [reply]
@dcacenabes I have also encountered this issue and have not found a good solution at the moment.
The easiest way to resolve it is to comment `RugbyTests` in `Package.swift` file and then, after closing Xcode, uncomment it and run the `make test` command.

As I understand it, this may be a bug, or we may need to move all the code to a basic target (non-executable) and test it there:
- https://github.com/apple/swift-package-manager/pull/3316
- https://forums.swift.org/t/unit-testing-executable-targets-with-swift-package-manager/44860

**Reply by dcacenabes** [reply]
Thanks for sharing! I might take a look at this at some point, although it is not a big blocker and currently has a workaround. Maybe the workaround could be added to the documentation of the project under a "known issues" section so that the entry barrier for contributors is lower.

**Reply by swiftyfinch** [reply]
Thank you! I've added it there: https://github.com/swiftyfinch/Rugby#-contribution.
